tokio/src/sync/broadcast.rs

```diff
@@ -1012,6 +1075,60 @@ impl<T> Drop for Sender<T> {
     }
 }
 
+impl<T> WeakSender<T> {
+    /// Tries to convert a `WeakSender` into a [`Sender`]. This will return `Some`
+    /// if there are other `Sender` instances alive and the channel wasn't
+    /// previously dropped, otherwise `None` is returned.
+    pub fn upgrade(&self) -> Option<Sender<T>> {
+        let mut tx_count = self.shared.num_tx.load(Acquire);
+
+        loop {
+            if tx_count == 0 {
+                // channel is closed so this WeakSender can not be upgraded
+                return None;
+            }
+
+            match self
+                .shared
+                .num_tx
+                .compare_exchange_weak(tx_count, tx_count + 1, AcqRel, Acquire)
```

Comment:
This is sufficient. When this succeeds, the operation is an increment for which relaxed is enough. When it fails, we might exit through the zero-check, so it needs acquire.
```suggestion
                .compare_exchange_weak(tx_count, tx_count + 1, Relaxed, Acquire)
```